packages/adblocker/src/engine/engine.ts

```diff
@@ -1174,16 +1178,32 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
 
     callerContext?: any | undefined;
   }): {
-    matches: {
-      filter: CosmeticFilter;
-      exception: CosmeticFilter | undefined;
-    }[];
+    matches: (
+      | {
+          filter: CosmeticFilter | undefined;
+          exception: CosmeticFilter | undefined;
+        }
+      | {
+          filter: undefined;
+          exception: NetworkFilter | undefined;
+        }
+    )[];
     allowGenericHides: boolean;
   } {
     domain ||= '';
 
-    let allowGenericHides = true;
-    let allowSpecificHides = true;
+    const matches: (
```

Comment:
cannot we copy `matchCosmeticFilters` return type?

Reply:
addressed in 7c8ad003c74ffa38081b477dd579ab144dc49983